packages/core/README.md

```diff
@@ -70,6 +70,44 @@ const governor = await governor.deployed();
 Note: this use case is not particularly common, but it is sometimes useful to have access to multiple abi versions
 side-by-side.
 
+## Deployment with Hardhat
+
+Here is a list of scripts you can execute:
+
+`yarn void:deploy`
+
+This will deploy your contracts on the in-memory hardhat network and exit, leaving no trace. Quickest way to ensure that deployments work as intended without consequences.
+
+`yarn hardhat deploy --network <NETWORK-NAME> --tags <TAGS>`
+
+Deploy all contracts to specified network. Requires a `CUSTOM_NODE_URL` HTTP(s) endpoint and a `MNEMONIC` to be set in environment. Available contract tags can be found in `/deploy` scripts, and available networks are found in the `networks` object within `hardhat.config.js`. Tags can be powerful, for example running `yarn hardhat deploy --tags Bridge` will only deploy the Bridge contract its dependencies (such as the Finder).
+
+`./scripts/hardhat/deployContracts.sh beacon-l2 <NETWORK-NAME>`
+
+Deploys contracts necessary to set up Beacon Oracle on L2 on the network, which would be used to deploy to Polygon for example.
+
+`./scripts/hardhat/deployContracts.sh beacon-l1 <NETWORK-NAME>`
+
+Deploys contracts necessary to set up Beacon Oracle on L1 on the network, which would be deployed on Mainnet for example.
+
+`./scripts/hardhat/verifyDeployedContracts.sh <NETWORK-NAME>`
+
+Verify contracts for selected network on Etherscan. Requires an `ETHERSCAN_API_KEY` to be set in environment. This script requires that the local `./core/deployments` has solc standard-input json files, which will be generated after running the `deploy` command.
+
+`yarn hardhat export --export-all ./networks/hardhat/deployed.json`
+
+Export deployed contract data such as ABI and addresses to `deployed.json` in order to make data available for external applications. For example, a newly deployed `Finder` address on Rinkeby can be imported via `deployed.4.rinkeby.contracts.Finder.address`.
+
+The following commands are implemented as [hardhat tasks](https://hardhat.org/guides/create-task.html) that make it easy to interact with deployed contracts via the CLI:
+
+`yarn hardhat register-deployer --network <NETWORK-NAME>`
+
+Registers the `deployer` account (as defined in the `namedAccounts` param in `hardhat.config.js`) with the deployed Registry for the network.
+
+`yarn hardhat setup-finder --registry --bridge --generichandler --network <NETWORK-NAME>`
+
+Sets specified contracts in the deployed Finder. More contracts available to be set can be found in the `core/scripts/hardhat/tasks/finder.js` script.
```

Comment:
Is the idea that most/all scripts that we currently have in scripts/ should be implemented as tasks in hardhat?

Reply:
Yeah that's what my crazy thought is long term. They could be ported pretty easily as scripts, which would make the CLI invocation a lot less verbose. The main benefit it seems of using tasks is you get access to the hardhat runtime environment which has a lot of nifty objects: https://hardhat.org/advanced/hardhat-runtime-environment.html

Reply:
Yep you can use HRE in js scripts